### PR TITLE
feat: remove request URI query params from emitted url.path

### LINKED ISSUE #214: Reduce cardinality of http.url, break up non-sensitive components into more attributes
Currently we capture the whole HTTP request URL and store the whole value as `http.request.url`. URLs can contains sensitive information that we need to be careful of, eg query parmeters and restful URLs (see below).

Investigate URL Shaper to parse HTTP request URLs and break into subcomponents. [URL shaper](https://github.com/honeycombio/urlshaper) is a tool that analyses and break up URLs into sub components to reduce cardinality and also reduces risk of capturing sensitive information. This issue track integrating this tools to analyse captures HTTP URLs.

### PR DESCRIPTION
**Pull request by robbkidd**
## Which problem is this PR solving?

- Closes #214 

## Short description of the changes

- add tests for path-only `url.path`
- use `url` package for URL parsing and emit only the path portion of the URI for `url.path`

## How to verify that this has the expected result

No more content after a # or a ? in a URL sent for `url.path`!

### CODE REVIEW

**Comment by robbkidd** [comment]
urlshaper does the job for removing parameters from request URIs. I stopped changing code at the point where it did what #214 described: no query params in the `url.path`. So at this point, both cardinality of `url.path` and the likelihood of sensitive information getting passed along have been reduced. Also, now we're following [the convention](https://opentelemetry.io/docs/specs/otel/common/url/) for `url.path` which should only be—like—the _path_.

We _could_ put them in `url.query` per semconv, but without more data munging, the values could still be sensitive.

urlshaper's Shape field is tempting, too, with its "query param values transformed into '?'" as output. There's no semconv-named field for that kind of value, though.

File: handlers/libhoney_event_handler.go [hunk 1]
```diff
@@ -155,7 +158,10 @@ func (handler *libhoneyEventHandler) handleEvent(event assemblers.HttpEvent) {
 		ev.AddField(string(semconv.HTTPRequestBodySizeKey), event.Request.ContentLength)
 		if handler.config.IncludeRequestURL {
 			requestURI = event.Request.RequestURI
-			ev.AddField(string(semconv.URLPathKey), requestURI)
+			urlShape, err := handler.urlParser.Parse(requestURI)
```
Comment (MikeGoldsmith) [inline handlers/libhoney_event_handler.go:161]: How do you feel about always adding the templated path and then using the new `IncludeRequesURL` config option to add in the original url?

Comment (robbkidd) [reply handlers/libhoney_event_handler.go]: Templated path is not something we've got yet without accepting regexes from the user to know how to parse out the variable bits of a path. 😞

**Comment by robbkidd** [comment]
Here's a go at emitting strictly the `url.path` (e.g. without query params) by using only stdlib's `url` package. No urlshaper needed yet.

**Review by MikeGoldsmith — APPROVED** [review]

**Review by JamieDanielson — APPROVED** [review]
keep your secrets

**Pull request merged**